test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs

```diff
@@ -39,85 +39,88 @@ public CustomSearchParamTests(HttpIntegrationTestFixture fixture, ITestOutputHel
-            _output.WriteLine("Reindex job is completed, it should have reindexed the Patient resources with foo");
+                reindexJobResult = await WaitForReindexStatus(reindexJobUri, "Completed");
+                _output.WriteLine($"Reindex job is completed, it should have reindexed the resources with {randomName}");
 
-            var floatParse = float.TryParse(
-                reindexJobResult.Resource.Parameter.FirstOrDefault(predicate => predicate.Name == "resourcesSuccessfullyReindexed").Value.ToString(),
-                out float resourcesReindexed);
+                var floatParse = float.TryParse(
+                    reindexJobResult.Resource.Parameter.FirstOrDefault(predicate => predicate.Name == "resourcesSuccessfullyReindexed").Value.ToString(),
+                    out float resourcesReindexed);
 
-            _output.WriteLine($"Reindex job is completed, {resourcesReindexed} Patient ressources Reindexed");
+                _output.WriteLine($"Reindex job is completed, {resourcesReindexed} resources Reindexed");
 
-            Assert.True(floatParse);
-            Assert.True(resourcesReindexed > 0.0);
+                Assert.True(floatParse);
+                Assert.True(resourcesReindexed > 0.0);
 
-            // When job complete, search for resources using new parameter
-            await ExecuteAndValidateBundle($"Patient?{searchParam.Code}:exact={patientName}", expectedPatient.Resource);
-
-            // Clean up new SearchParameter
-            await DeleteSearchParameterAndVerify(searchParamPosted.Resource);
+                // When job complete, search for resources using new parameter
+                await ExecuteAndValidateBundle($"Appointment?{searchParam.Code}=noshow&_tag={tag.Code}", expectedAppointment.Resource);
+            }
+            catch (FhirException ex) when (ex.StatusCode == HttpStatusCode.BadRequest && ex.Message.Contains("not enabled"))
+            {
+                Skip.If(!_fixture.IsUsingInProcTestServer, "Reindex is not enabled on this server.");
+                return;
+            }
+            catch (Exception e)
+            {
+                _output.WriteLine($"Exception: {e.Message}");
+                _output.WriteLine($"Stack Trace: {e.StackTrace}");
+                throw;
+            }
+            finally
+            {
+                // Clean up new SearchParameter
+                await DeleteSearchParameterAndVerify(searchParamPosted?.Resource);
+            }
         }
 
         [SkippableFact]
         public async Task GivenASearchParam_WhenUpdatingParam_ThenResourcesIndexedWithUpdatedParam()
         {
-            var patientName = Guid.NewGuid().ToString().ComputeHash().Substring(28).ToLower();
-            var patient = new Patient { Name = new List<HumanName> { new HumanName { Family = patientName } } };
+            var randomName = Guid.NewGuid().ToString().ComputeHash().Substring(28).ToLower();
```

Comment:
>Substring(28) [](start = 69, length = 13)

just curious, why it's substring 28 here and substring 14 in test above?

Reply:
No good reason :).